src/test/java/org/fogbowcloud/manager/core/processors/SpawningProcessorTest.java

```diff
@@ -60,243 +55,279 @@ public void setUp() {
+        Mockito.when(this.tunnelingService.getExternalServiceAddresses(Mockito.anyString()))
+                .thenReturn(new HashMap<>());
+        Mockito.when(this.sshConnectivity
+                .checkSSHConnectivity(Mockito.any(SshTunnelConnectionData.class)))
+                .thenReturn(false);
 
-        when(this.tunnelingService.getExternalServiceAddresses(anyString())).thenReturn(new HashMap<>());
-        
-        when(this.sshConnectivity.checkSSHConnectivity(any(SshTunnelConnectionData.class))).thenReturn(false);
-        
         this.thread = new Thread(this.spawningProcessor);
         this.thread.start();
         this.thread.sleep(500);
 
-        assertNotNull(this.spawningOrderList.getNext());
-        assertNull(this.fulfilledOrderList.getNext());
+        // verify
+        Assert.assertNotNull(this.spawningOrderList.getNext());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
-    
+
+    // test case: When running thread in SpawningProcessor without the external addresses of the
+    // reverse tunnel, the method processSpawningOrder() must not change OrderState to Fulfilled and
+    // must remain in Spawning list.
     @SuppressWarnings("static-access")
     @Test
     public void testRunProcessComputeOrderWithoutExternalAddresses() throws Exception {
+
+        // set up
         Order order = spyComputeOrder();
         order.setOrderState(OrderState.SPAWNING);
         this.spawningOrderList.addItem(order);
-        
+
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.READY);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
-        doReturn(orderInstance).when(this.cloudConnector).getInstance(any(Order.class));
+        // exercise
+        Mockito.doReturn(orderInstance).when(this.cloudConnector)
+                .getInstance(Mockito.any(Order.class));
+        Mockito.when(this.tunnelingService.getExternalServiceAddresses(Mockito.anyString()))
+                .thenReturn(null);
 
-        when(this.tunnelingService.getExternalServiceAddresses(anyString())).thenReturn(null);
-        
         this.thread = new Thread(this.spawningProcessor);
         this.thread.start();
         this.thread.sleep(500);
 
-        assertNotNull(this.spawningOrderList.getNext());
-        assertNull(this.fulfilledOrderList.getNext());
+        // verify
+        Assert.assertNotNull(this.spawningOrderList.getNext());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
-    
+
+    // test case: When running thread in the SpawningProcessor and the OrderType is not a
+    // Compute, the processSpawningOrder() method must immediately change the OrderState to
+    // Fulfilled by adding in that list, and removed from the Spawning list.
     @SuppressWarnings("static-access")
     @Test
     public void testRunProcessWhenOrderTypeIsNotCompute() throws Exception {
+
+        // set up
```

Comment:
how about the other types? at least, add a fixme to remember we need to add the others.